src/skmatter/metrics/_pairwise.py

```diff
@@ -0,0 +1,230 @@
+from typing import Union
+
+import numpy as np
+from sklearn.metrics.pairwise import (
+    _euclidean_distances,
+    check_array,
+    check_pairwise_arrays,
+)
+
+
+def pairwise_euclidean_distances(
+    X, Y=None, *, Y_norm_squared=None, squared=False, X_norm_squared=None, cell=None
+):
+    """
+    Compute the pairwise distance matrix between each pair from a vector array X and Y.
+
+    .. math::
+        d_{i, j} = \\sqrt{\\sum_{k=1}^n (x_{i, k} - y_{j, k})^2}
+
+    For efficiency reasons, the euclidean distance between a pair of row
+    vector x and y is computed as::
+
+        dist(x, y) = sqrt(dot(x, x) - 2 * dot(x, y) + dot(y, y))
+
+    This formulation has two advantages over other ways of computing distances. First,
+    it is computationally efficient when dealing with sparse data. Second, if one
+    argument varies but the other remains unchanged, then `dot(x, x)` and/or `dot(y, y)`
+    can be pre-computed.
+
+    However, this is not the most precise way of doing this computation, because this
+    equation potentially suffers from "catastrophic cancellation". Also, the distance
+    matrix returned by this function may not be exactly symmetric as required by, e.g.,
+    ``scipy.spatial.distance`` functions.
+
+    Read more in the :ref:`User Guide <metrics>`.
+
+    Parameters
+    ----------
+    X : {array-like, sparse matrix} of shape (n_samples_X, n_components)
+        An array where each row is a sample and each column is a component.
+
+    Y : {array-like, sparse matrix} of shape (n_samples_Y, n_components), \
+            default=None
+        An array where each row is a sample and each column is a component.
+        If `None`, method uses `Y=X`.
+
+    Y_norm_squared : array-like of shape (n_samples_Y,) or (n_samples_Y, 1) \
```

Comment:
You don't need empty lines between the arguments. I don't why this was done in the past. I also removed them from other functions in #227.

Can you do this for your the doc strings as well?

```suggestion
    X : {array-like, sparse matrix} of shape (n_samples_X, n_components)
        An array where each row is a sample and each column is a component.
    Y : {array-like, sparse matrix} of shape (n_samples_Y, n_components), \
            default=None
        An array where each row is a sample and each column is a component.
        If `None`, method uses `Y=X`.
    Y_norm_squared : array-like of shape (n_samples_Y,) or (n_samples_Y, 1) \
```